### PR TITLE
Backend now connects to database. Users can be created.

### PR DESCRIPTION
**Pull request by Mahatmasamatman**
There is now a db.js file that handles connection to the database.
Database credentials have been omitted using .gitignore because this is a public repository.

There are currently 2 routes, /users and /auth route. Both routes are imported into server.js which is our main entry file.

Users route validates the incoming data that is used to register a user, and if the data is valid, and the users email is not taken, the entry gets added into the database.

Ran Prettier through all of the files,

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
+import users from './api/users.js';
+
+connectDB();
 
 const app = express();
 
+//Init Middleware - it allows routes to access the data in req.body
+app.use(express.json({ extended: false }));
+
 app.get('/', (req, res) => res.send('API Runnnig'));
```
Comment (frle10) [inline server.js:13]: Typo "Runnnig"

File: api/users.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import express from 'express';
+import { check, validationResult } from 'express-validator';
+import bcrypt from 'bcryptjs';
+import User from '../models/User.js';
+
+const router = express.Router();
```
Comment (frle10) [inline api/users.js:6]: I can see that the router is instantiated both in users.js and auth.js. Is this something that's necessary or can the router be instantiated once and then reused wherever it needs to be?

I never worked with pure Express so I'm not sure how this works completely.

**Comment by frle10** [comment]
The code is cleanly written and well formatted. Nice to see you're using MongoDB, since you've always been using PostgresSQL in the past. Tbh, PostgresSQL is much better in every way and Mongo is usually used for BigData use cases, but learning other databases is a very nice addition to your knowledge library. :)

File: api/users.js (hunk 2 above)
Comment (Mahatmasamatman) [reply api/users.js:6]: The way I see it, no (and I did try it just to see what would happen, and in that case one route overrides the other). 

express.Router() creates a router object, where we define the routing for specific http methods, and then we pass it to app.use

```
const app = express();

app.use('/api/users', users);
app.use('/api/auth', auth);
```

which 'uses' the router when the path of the request matches the specified string.
In this case it wouldn't even make sense for both paths to use the same router.

File: server.js (hunk 1 above)
Comment (frle10) [reply server.js]: This has not been resolved.

File: server.js [hunk 3]
```diff
@@ -1,9 +1,21 @@
...
+
+connectDB();
 
 const app = express();
 
-app.get('/', (req, res) => res.send('API Runnnig'));
+//Init Middleware - it allows routes to access the data in req.body
+app.use(express.json({ extended: false }));
+
+app.get('/', (req, res) => res.send('API Runnning'));
```
Comment (frle10) [inline server.js:13]: Typo again :)

File: server.js (hunk 1 above)
Comment (Mahatmasamatman) [reply server.js]: image

Comment (Mahatmasamatman) [reply server.js]: This is the latest commit

File: server.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ const app = express();
 //Init Middleware - it allows routes to access the data in req.body
 app.use(express.json({ extended: false }));
 
-app.get('/', (req, res) => res.send('API Runnning'));
+app.get('/', (req, res) => res.send('API is running.'));
```
Comment (frle10) [inline server.js:13]: Absolute perfection! Approve coming your way :)

**Review by frle10 — APPROVED** [review]

**Pull request merged**